### ISSUE TITLE
NU1605

### ISSUE THREAD

**Issue by petermorlion** [Azure/amqpnetlite#275]
When we reference this library and build it in JetBrains Rider, we get a NU1605 warning. If warnings are set as error this breaks. 

The warning is because the AMQPNetLite library references System.Net.NameResolution 4.3.0 and 4.0.0, albeit via other assemblies:

MyAssembly -> AMQPNetLite 2.1.2 -> AMQPNetLite.Core 2.1.2 -> System.Net.Sockets 4.1.0 -> runtime.win.System.Net.Sockets 4.3.0 -> System.Net.NameResolution >= 4.3.0

and

MyAssembly -> AMQPNetLite 2.1.2 -> AMQPNetLite.Core 2.1.2 -> System.Net.NameResolution 4.0.0

[This issue](https://github.com/dotnet/core/issues/907) states that it can be fixed by referencing the more recent assembly directly in `MyAssembly`. But a more permanent fix would be for the library to reference the newer one. So I think AMQPNetLite.Core should reference System.Net.NameResolution 4.3.0.

If you want, I can create a PR for this. But I wanted to hear your opinion first.

**Comment by xinchen10** [comment]
@petermorlion it would be great if you can create a PR for this. The current build tools (1.0) doesn't give me this warning.

**Comment by billpoole-mi** [comment]
We have the same problem every time we "Publish" our app from Visual Studio for a specific platform (i.e. for any platform other than "Portable". We therefore have to ignore NU1605 to be able to Publish for deployment. It would be much appreciated if this could be fixed.

**Comment by xinchen10** [comment]
Also after publishing runtime.win.System.Net.Sockets 4.1.0 got installed under nuget packages folder, not 4.3.0. I am not sure if this is related to the dotnet CLI or something else.

**Comment by billpoole-mi** [comment]
We are publishing for the debian-x64 platform using Visual Studio 15.7.1 using .NET Core SDK 2.1.200. However we have the same NU1605 errors reported when publishing for win-x64 as well. We've also experienced the same problem from when we first started using Amqp.Net Lite in Februrary with whatever versions of Visual Studio and the .NET Core SDK have been released through that period.

Looking into it in more detail, at least one source of our conflict is between Amqp.Net Lite and Npgsql (the PostgreSQL .NET client). I've attached a Visual Studio solution to illustrate as requested. Thinking about it some more, I suspect the issue is that there is no .NET Standard 2.0 specific build of Amqp.Net Lite, which therefore is based upon earlier versions of the assemblies that the latest version of Npgsql (based on .NET Standard 2.0) requires.

So (possibly answering my own question), is the solution to add a .NET Standard 2.0 build/distribution for Amqp.Net Lite?

[PackageDowngrade.zip](https://github.com/Azure/amqpnetlite/files/2079614/PackageDowngrade.zip)
